lib/mail/renderers/rfc_2822.ex

```diff
@@ -251,30 +251,83 @@ defmodule Mail.Renderers.RFC2822 do
     |> String.pad_leading(2, "0")
   end
 
-  defp reorganize(%Mail.Message{multipart: true} = message) do
+  defp split_attachment_parts(message) do
+    Enum.reduce(message.parts, [[], [], []], fn part, [texts, mixed, inlines] ->
```

Comment:
Would it not be better to work with a tuple of lists `{[], [], []}` to be more explicit about the number of lists being generated?